platform/android/src/native_map_view.cpp

```diff
@@ -827,47 +827,44 @@ void NativeMapView::addLayerAt(JNIEnv& env, jlong nativeLayerPtr, jni::jint inde
 /**
  * Remove by layer id.
  */
-jni::Local<jni::Object<Layer>> NativeMapView::removeLayerById(JNIEnv& env, const jni::String& id) {
-    std::unique_ptr<mbgl::style::Layer> coreLayer = map->getStyle().removeLayer(jni::Make<std::string>(env, id));
-    if (coreLayer) {
-        return LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
-    } else {
-        return jni::Local<jni::Object<Layer>>();
-    }
-}
 
 /**
  * Remove layer at index.
  */
-jni::Local<jni::Object<Layer>> NativeMapView::removeLayerAt(JNIEnv& env, jni::jint index) {
+jni::jboolean NativeMapView::removeLayerAt(JNIEnv& env, jni::jint index) {
     auto layers = map->getStyle().getLayers();
 
     // Check index
     int numLayers = layers.size() - 1;
     if (index > numLayers || index < 0) {
         Log::Warning(Event::JNI, "Index out of range: %i", index);
-        return jni::Local<jni::Object<Layer>>();
+        return jni::jni_false;
     }
 
     std::unique_ptr<mbgl::style::Layer> coreLayer = map->getStyle().removeLayer(layers.at(index)->getID());
     if (coreLayer) {
-        return LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
-    } else {
-        return jni::Local<jni::Object<Layer>>();
+        jni::Local<jni::Object<Layer>> layerObj =
+                LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
+        mbgl::android::Layer *layer = reinterpret_cast<mbgl::android::Layer *>(layerObj.get());
+        layer->setLayer(std::move(coreLayer));
```

Comment:
@tobrun could you ask someone with more knowledge of c++ how this should be done . Lines 846 -848 are not correct.